rules/list-item.js

```diff
@@ -102,12 +102,24 @@ function validateList(list, file) {
 			continue;
 		}
 
-		const [link, ...description] = paragraph.children;
-
-		if (link.type === 'text') {
+		if (paragraph.children[0].type === 'text') {
 			continue;
 		}
 
+		let [link, ...description] = paragraph.children;
+
+		// Might have children like {image} {text} {link} - {descrition}
+		// Keep discarding elements until we find a link
+		//                                       or linkReference
+		if (!/link/.test(link.type)) {
```

Comment:
Compare explicitly. Doesn't make sense to use regex here.

Reply:
besides linkReference and link, I'm not sure if there are other link items in markdown -- figured any type matching "link" should get to pass
`link.type !== "linkReference" && linkk.type !== "link"` is a mouth full :)

Reply:
You could use `Array#includes`.

---

> besides linkReference and link, I'm not sure if there are other link items in markdown

Look at the docs for the Markdown parser.